### PR TITLE
[gtest] rewrite version patch

### LINKED ISSUE #40: gtest errors
When using gtest we get the following error:
```
CMake Warning at /usr/src/gmock/CMakeLists.txt:41 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


CMake Warning at /usr/src/googletest/CMakeLists.txt:51 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


-- Found Python: /usr/bin/python3.10 (found version "3.10.5") found components: Interpreter
-- Found Threads: TRUE
CMake Error at /usr/src/googletest/CMakeLists.txt:127 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/googletest/CMakeLists.txt:129 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/gmock/CMakeLists.txt:99 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/gmock/CMakeLists.txt:102 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
```

It seems like the version patch we have is not doing it's job properly?
I'll prob look into this at some point, but wanted to jot it down so if anyone else inclined can resolve the issue.

Bug report on bugs.archlinux.org: https://bugs.archlinux.org/task/68440?string=gtest&project=5&type%5B0%5D=&sev%5B0%5D=&pri%5B0%5D=&due%5B0%5D=&reported%5B0%5D=&cat%5B0%5D=&status%5B0%5D=open&percent%5B0%5D=&opened=&dev=&closed=&duedatefrom=&duedateto=&changedfrom=&changedto=&openedfrom=&openedto=&closedfrom=&closedto=

### PR DESCRIPTION
**Pull request by acxz**
Resolves #40 

It turns out the patch that we had before is a bit old. I redid the patch to properly get the GOOGLETEST_VERSION variable assigned.

The problem itself comes up due to how user libraries include gtest/gmock, see here: https://github.com/google/googletest/issues/2950 for a more thorough explanation.

Sometimes people will build the gtest or gmock subdirectory instead of building from the root with a say `-DBUILD_GMOCK=OFF`, which means the `GOOGLETEST_VERSION` does not get propagated down.

Basically I don't think it is the responsibility of us to patch this, but considering that we had a version patch in the past, I think its fine to update the patch properly. And many many projects still build from the subdirectories instead of the root directory.

The second `sed` is not needed with the newest version as the version in the root CMakelists.txt is correct.

### CODE REVIEW

**Comment by acxz** [comment]
@anthraxx friendly ping

**Comment by acxz** [comment]
@anthraxx can you provide closure to this PR?

**Comment by acxz** [comment]
closing this in favor of #45
